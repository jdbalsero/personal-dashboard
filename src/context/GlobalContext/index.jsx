import React, { createContext, useState, useEffect } from 'react';

const GlobalContext = createContext();

function PageProvider(props) {
    const [activePage, setActivePage] = useState(() => {
        // Initialize activePage from localStorage or set a default value
        const savedActivePage = localStorage.getItem('activePage');
        return savedActivePage || 'Home';
    });

    const [activeBlur, setActiveBlur] = useState(() => {
        // Initialize activeBlur from localStorage or set a default value
        const savedActiveBlur = localStorage.getItem('activeBlur');
        return savedActiveBlur || '';
    });


    useEffect(() => {
        // Save state values to localStorage whenever they change
        localStorage.setItem('activePage', activePage);
        localStorage.setItem('activeBlur', activeBlur);
    }, [activePage, activeBlur]);

    return (
        <GlobalContext.Provider
            value={{
                activePage,
                setActivePage,
                activeBlur,
                setActiveBlur,
            }}
        >
            {props.children}
        </GlobalContext.Provider>
    );
}

export { GlobalContext, PageProvider };


