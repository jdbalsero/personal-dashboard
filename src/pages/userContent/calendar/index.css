.event-add-button{
    display: flex;
    width: 15%;
    height: 35px;
    justify-content: center;
    align-items: center;
    border-radius: 8px;
    border-color: transparent;
    font-weight: bold;
    color: #FFF;
    background: var(--main-blue-color);
    cursor: pointer;
    margin-bottom: 6px;
}