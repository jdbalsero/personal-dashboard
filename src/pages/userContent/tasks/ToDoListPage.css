.todo-general-container{
    padding: 0 4%;
}

.todo-title{
    font-style: normal;
    font-size: 2em;
    letter-spacing: -0.01em;
    width: 100%;
    display: flex;
    justify-content: left;
    align-items: center;
    color: var(--main-petroleum-color);
    margin:0 15px;
}

.todo-subtitle{
    font-style: normal;
    font-size: 1em;
    letter-spacing: -0.01em;
    width: 100%;
    display: flex;
    justify-content: left;
    align-items: center;
    color: var(--main-petroleum-color);
    margin:0 15px;
}

.list-todo{
    height: 320px;
    overflow-y: scroll;
    padding-right: 10px;
}

.list-todo::-webkit-scrollbar {
    width: 8px;
}

.list-todo::-webkit-scrollbar-thumb {
    background-color: #9D9D9D;
    border-radius: 6px;
}

.list-todo::-webkit-scrollbar-track {
    background-color: #f1f1f100;
}

.todo-item{
    border-radius: 10px;
    background: #FFF;
    box-shadow: 3px 3px 3px 3px rgba(0, 0, 0, 0.25);
    margin-top: 20px;
    padding: 10px;
}

.icon{
    justify-content: center;
}

.icon-color2{
    color: var(--main-petroleum-color);
}

.icon-color{
    color:#000;
}

.todo-add-button-container{
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: 25px 0px;
    left: 0;
    bottom: 0;
    color: white;
    text-align: center;
}
    

.todo-add-button{
    display: flex;
    width: 20%;
    height: 45px;
    padding: 8px 15px;
    justify-content: center;
    align-items: center;
    gap: 10px;
    flex-shrink: 0;
    border-radius: 8px;
    border-color: transparent;
    font-weight: bold;
    color: #FFF;
    background: var(--main-petroleum-color);
    cursor: pointer;
}

.inputTodo{
    height:45px;
    width: 78%;
    border-radius: 10px;
    background: #FFF;
    box-shadow: 3px 3px 3px 3px rgba(0, 0, 0, 0.25);
}

.Return_Button {
    display: flex;
    align-items: flex-start;
    padding: 20px;
    gap: 10px;
    width: 30px;
    height: 37px;
    border-radius: 7px;
    order: 1;
    flex-grow: 0;
    border: none;
    cursor: pointer;
    font-family: 'aeonik';
    font-style: normal;
    font-weight: bolder;
    font-size: 14px;
    line-height: 17px;
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    letter-spacing: -0.01em;
    color: #FFFFFF;
    margin-bottom: 10px;
}