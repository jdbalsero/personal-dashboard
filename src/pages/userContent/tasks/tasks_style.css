.list-task{
    height: 320px;
    overflow-y: scroll;
    padding-right: 10px;
}

.list-task::-webkit-scrollbar {
    width: 8px;
}

.list-task::-webkit-scrollbar-thumb {
    background-color: #9D9D9D;
    border-radius: 6px;
}

.list-task::-webkit-scrollbar-track {
    background-color: #f1f1f100;
}

.task-item{
    border-radius: 10px;
    background: #FFF;
    box-shadow: 3px 3px 3px 3px rgba(0, 0, 0, 0.25);
    margin-top: 20px;
    padding: 10px;
}

.icon-color{
    color:#000;
}

.todo-add-button-container{
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: 25px 0px;
    left: 0;
    bottom: 0;
    color: white;
    text-align: center;
}
    

.todo-add-button{
    display: flex;
    width: 20%;
    height: 45px;
    padding: 8px 15px;
    justify-content: center;
    align-items: center;
    gap: 10px;
    flex-shrink: 0;
    border-radius: 8px;
    border-color: transparent;
    font-weight: bold;
    color: #FFF;
    background: var(--main-blue-color);
    cursor: pointer;
}

.inputTodo{
    height:45px;
    width: 78%;
    border-radius: 10px;
    background: #FFF;
    box-shadow: 3px 3px 3px 3px rgba(0, 0, 0, 0.25);
}

.task-checkbox-container{
    justify-content: end;
}

.linethrough{
    text-decoration: line-through;
}