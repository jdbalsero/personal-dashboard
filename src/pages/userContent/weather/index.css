.weather-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
  
  .weather-box {
    display: flex;
    align-items: center;
    justify-content: space-between;
    gap:25px;
    background-color: #FFFFFF;
    border-radius: 10px;
    box-shadow: 0px 5px 10px rgba(0, 0, 0, 0.1);
    padding: 20px;
  }

  .temperature {
    font-size: 50px;
    margin: 0;
  }
  
  .secondary-text {
    font-size: 20px;
    margin: 0;
  }

  .details-text {
    font-size: 18px;
    margin: 0;
  }

  .icon-text{
    display:flex;
    align-items: center;
    gap:2px;
    margin:0;
  }