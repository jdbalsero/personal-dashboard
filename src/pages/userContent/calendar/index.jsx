import React, { useState, useEffect, useContext } from 'react';
import { Popup } from '../../../components/Shared/Popup';
import Backdrop from '@mui/material/Backdrop';
import CircularProgress from '@mui/material/CircularProgress';
import { GlobalContext } from "../../../context/GlobalContext";
import { ContentHeader } from '../../../components/Shared/ContentHeader';
import { CalendarBox } from '../../../components/Shared/Calendar';
import { PopupCreateEvent } from './PopupCreateEvent';
import { Button } from "../../../components/Shared/Button";
import Swal from 'sweetalert2';
import withReactContent from 'sweetalert2-react-content';
import './index.css';
const MySwal = withReactContent(Swal);


function CalendarClient() {
    const [events, setEvents] = useState([]);
    const [isPopupOpen, setIsPopupOpen] = useState(false);
    const { setActiveBlur } = useContext(GlobalContext);

    const fetchEvents = async () => {
        try {
            let dataEvents = JSON.parse(localStorage.getItem("Events")) === null ? [] : JSON.parse(localStorage.getItem("Events"));
            const eventsCalendar = await dataEvents.map((item, index) => {
                return {
                    id: index,
                    title: item.title,
                    start: new Date(item.start),
                    end: new Date(item.end),
                    description: item.description,
                }
            });
            setEvents(eventsCalendar);
        } catch (error) {
            console.error(error);
            setActiveBlur(true);
            MySwal.fire({
                icon: 'error',
                title: 'Oops ...',
                text: 'There was a unexpected error retrieving the tasks, please contact support.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
        }
    };

    useEffect(() => {
        fetchEvents();
    }, []);

    const createButtonClick = () => {
        setIsPopupOpen(true);
        setActiveBlur(true);
    }

    const handleAddEvent = (data) => {
        try{
            events.push(data);
            localStorage.setItem("Events", JSON.stringify(events));
            setActiveBlur(true);
            MySwal.fire({
                icon: 'success',
                title: 'Success',
                text: 'Event Saved.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
            closePopup();
            fetchEvents();
        }catch(err){
            console.error(err);
            setActiveBlur(true);
            MySwal.fire({
                icon: 'error',
                title: 'Oops ...',
                text: 'There was a unexpected error adding the event, please contact support.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
        }
    }

    const closePopup = () => {
        setIsPopupOpen(false);
        setActiveBlur(false);
    }

    return <React.Fragment>
        <div className="calendar">
            <div className="calendar-header">
                <ContentHeader text="Calendar" />
            </div>
            <div className="calendar-body">
                { events !== undefined ? 
                    <>
                        <Popup isOpen={isPopupOpen} onClose={closePopup} modalClass="popup-info-task">
                            <PopupCreateEvent
                                handleAddEvent = {handleAddEvent}
                                closePopup = {closePopup}
                            />
                        </Popup>
                        <Button
                            buttonClassName="event-add-button"
                            onClick={createButtonClick}
                            children="Add Event"
                        />
                        <CalendarBox events={events} />
                    </>
                    :
                    <Backdrop
                        sx={{ color: '#fff', zIndex: (theme) => theme.zIndex.drawer + 1 }}
                        open={true}
                    >
                        <CircularProgress style={{ height: "50px", width: "100px" }} color="inherit" />
                    </Backdrop>
                }
            </div>
        </div>
    </React.Fragment>
}

export { CalendarClient };