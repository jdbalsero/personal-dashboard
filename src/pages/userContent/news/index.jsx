import React, { useEffect, useState, useContext } from 'react';
import { GlobalContext } from '../../../context/GlobalContext';
import { ContentHeader } from '../../../components/Shared/ContentHeader'
import { TableNews } from './TableNews';
import TextField from '@mui/material/TextField';
import Autocomplete from '@mui/material/Autocomplete';
import './news.css';
import Swal from 'sweetalert2';
import withReactContent from 'sweetalert2-react-content';
const MySwal = withReactContent(Swal);

function NewsPage() {
    const Categories = ['General', 'World', 'Business', 'Entertainment', 'Science', 'Sports', 'Health', 'Technology'];
    const { setActiveBlur } = useContext(GlobalContext);
    const [news, setNews] = useState([])
    const [value, setValue] = React.useState(Categories[0]);
    const [inputValue, setInputValue] = React.useState('');

    const showNews = async (category=null) => {
        const API_BASE_URL = category === null ? 
        `https://gnews.io/api/v4/top-headlines?category=general&lang=en&apikey=${process.env.REACT_APP_GNEWSAPI_KEY}`
        : `https://gnews.io/api/v4/top-headlines?lang=en&category=${category}&apikey=${process.env.REACT_APP_GNEWSAPI_KEY}`;
        try {
            const response = await fetch(API_BASE_URL).then(response => {
              return response;
            });
            if (!response.ok) {
                throw new Error('Error en la solicitud');
            }
            let data = await response.json();
            setNews(data.articles);
        } catch (error) {
            console.error(error);
            setActiveBlur(true);
            MySwal.fire({
                icon: 'error',
                title: 'Oops ...',
                text: 'There was an error searching news, please check your filter selection and try again.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
        }
      }

    useEffect(() => {
        showNews();
    }, [])

    return <div className="data-news-container">
        <ContentHeader text="News" />
        <div className="search-section">
            <Autocomplete
                disablePortal
                value={value}
                onChange={(event, newValue) => {
                    setValue(newValue);
                    showNews(String(newValue).toLowerCase());
                }}
                inputValue={inputValue}
                onInputChange={(event, newInputValue) => {
                    setInputValue(newInputValue);
                }}
                id="controllable-states-demo"
                className='input-search'
                options={Categories}
                sx={{ width: 300 }}
                renderInput={(params) => <TextField {...params} label="News Category" />}
            />
        </div>
        <TableNews data={news} fetchData={showNews} />
    </div>
}

export { NewsPage }