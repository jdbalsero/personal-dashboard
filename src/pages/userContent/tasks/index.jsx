import React, { useContext, useState, useEffect } from "react";
import { GlobalContext } from '../../../context/GlobalContext';
import { Button } from "../../../components/Shared/Button";
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Swal from 'sweetalert2';
import withReactContent from 'sweetalert2-react-content';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import Paper from '@mui/material/Paper';
import { styled } from '@mui/material/styles';
import IconButton from '@mui/material/IconButton';
import DeleteIcon from '@mui/icons-material/Delete';
import FormControlLabel from '@mui/material/FormControlLabel';
import OutlinedInput from '@mui/material/OutlinedInput';
import Checkbox from '@mui/material/Checkbox';
import { ContentHeader } from '../../../components/Shared/ContentHeader'
import './tasks_style.css';
const MySwal = withReactContent(Swal);

const Item = styled(Paper)(({ theme }) => ({
    backgroundColor: theme.palette.mode === 'dark' ? '#1A2027' : '#fff',
    ...theme.typography.body2,
    padding: theme.spacing(4),
    textAlign: 'center',
    color: theme.palette.text.secondary,
  }));
  

function TasksPage() {
    const [todolist_tasks, setToDoListTasks] = useState([]);
    const { setActiveBlur } = useContext(GlobalContext)
    const [inputValue, setInputValue] = useState('');

    const handleChange = (event) => {
        setInputValue(event.target.value);
    };

    const handleDeleteClick = (todo_id) => {
        setActiveBlur(true);
        MySwal.fire({
            icon:'warning',
            title: `Are you Sure?`,
            text: 'Do you want to delete this to do task ?',
            showDenyButton: true,
            showCancelButton: false,
            confirmButtonText: 'Yes, I want to delete it.',
            denyButtonText: `No, I want to keep it`,
            backdrop: true,
            customClass: {
                popup: 'popup-sweet',
                title: 'title-sweet',
                htmlContainer: 'text-sweet',
                confirmButton: 'confirm-button-sweet',
                denyButton: 'deny-button-sweet',
            }
        }).then((result) => {
            if (result.isConfirmed) {
                todolist_tasks.splice(todo_id,1);
                localStorage.setItem("Tasks", JSON.stringify(todolist_tasks));
                setActiveBlur(true);
                MySwal.fire({
                    icon: 'success',
                    title: 'Success',
                    text: 'To Do Task Deleted.',
                    showConfirmButton: false,
                    timer: 1700,
                    backdrop: true,
                    customClass: {
                        popup: 'popup-sweet',
                        title: 'title-sweet',
                        htmlContainer: 'text-sweet',
                        confirmButton: 'confirm-button-sweet',
                        denyButton: 'deny-button-sweet',
                    }
                }).finally(() => {setActiveBlur(false)});
                fetchData();
            }
        }).finally(() => {setActiveBlur(false)});
    };

    const handleAddToDoTask = async() => {
        try{
            todolist_tasks.push({description: inputValue, status:"Active"});
            localStorage.setItem("Tasks", JSON.stringify(todolist_tasks));
            setActiveBlur(true);
            MySwal.fire({
                icon: 'success',
                title: 'Success',
                text: 'To Do List Task Saved.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
            setInputValue('');
            fetchData();
        }catch(err){
            console.error(err);
            setActiveBlur(true);
            MySwal.fire({
                icon: 'error',
                title: 'Oops ...',
                text: 'There was a unexpected error adding the new task, please contact support.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
        }
    }

    const handleUpdateStatus = async (id, value) => {
        try{
            todolist_tasks[id].status = value ? "Completed":"Active";
            localStorage.setItem("Tasks", JSON.stringify(todolist_tasks));
            fetchData();
        }catch(err){
            console.error(err);
            setActiveBlur(true);
            MySwal.fire({
                icon: 'error',
                title: 'Oops ...',
                text: 'There was a unexpected error updating the task, please contact support.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
        }
    }

    const fetchData = async () => {
        try {
            let newRows = JSON.parse(localStorage.getItem("Tasks")) === null ? [] : JSON.parse(localStorage.getItem("Tasks"));
            setToDoListTasks(newRows);
        } catch (error) {
            console.error(error);
            setActiveBlur(true);
            MySwal.fire({
                icon: 'error',
                title: 'Oops ...',
                text: 'There was a unexpected error retrieving the tasks, please contact support.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
        }
    };

    const handleKeyDown = (e) => {
        if (e.key === 'Enter') {
          handleAddToDoTask();
        }
      }

    useEffect(() => {
        fetchData();
    }, []);

    return (
        <React.Fragment>
            <div className="todo-general-container">
                <ContentHeader text="Tasks" />
                {todolist_tasks.length > 0 ? 
                    <div className="list-task">
                        <List sx={{ width: '100%' }}>
                            {todolist_tasks.map((option, index) => {
                                const labelId = `item-list-label-${option}`;
                                return (
                                <div key={index}>
                                    <ListItem className="task-item" disablePadding secondaryAction={
                                        <IconButton edge="end" aria-label="delete" onClick={() => {handleDeleteClick(index);}}>
                                            <DeleteIcon className="icon-color"/>
                                        </IconButton>
                                    }>
                                        <ListItemIcon className="task-checkbox-container">
                                            <FormControlLabel
                                                control={
                                                    <Checkbox
                                                        checked={option.status === "Active" ? false : true}
                                                        style ={{
                                                            color: "#5873FE",
                                                        }}
                                                        onChange={e => handleUpdateStatus(index, e.target.checked)}
                                                    />
                                                }
                                            />
                                        </ListItemIcon>
                                        <ListItemText id={labelId} primary={option.description} className={option.status === "Completed" ? 'linethrough' : ''}  secondary={`Status: ${option.status}`}/>
                                    </ListItem>
                                </div>
                                );
                            })}
                        </List>
                    </div>
                : <Box sx={{ width: '100%' }} style={{ display: 'flex', justifyContent: 'center', margin:'17px' }}>
                        <Typography
                            sx={{ display: 'inline' }}
                            component="span"
                            variant="body2"
                            color="text.primary"
                        >
                            You don't have any task in the list
                        </Typography>
                    </Box>
                }
                <div className="todo-add-button-container">
                    <OutlinedInput className="inputTodo" type="text" placeholder="Type the task you need" value={inputValue} onChange={handleChange} onKeyDown={handleKeyDown} />
                    <Button
                        buttonClassName="todo-add-button"
                        onClick={() => handleAddToDoTask() }
                        children="Add Task"
                    />
                </div>
            </div>
        </React.Fragment>
    );
}

export { TasksPage };