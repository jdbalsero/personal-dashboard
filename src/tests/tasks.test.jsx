import { fireEvent, render, screen } from '@testing-library/react';
import { TasksPage} from '../pages/userContent/tasks';
import { PageProvider } from '../context/GlobalContext';
import React from 'react';
import { act } from 'react-dom/test-utils';


describe('<TasksPage /> Tests', () => {

    test('Match Snapshot', () => {
        const { container } = render(
        <PageProvider>
            <TasksPage />
        </PageProvider>);
        expect( container ).toMatchSnapshot();
    });

    test('Show Title of Page', () => {
        render(
            <PageProvider>
                <TasksPage />
            </PageProvider>
        );
        expect( screen.getByText("Tasks") ).toBeTruthy();
    });

    test('Show Empty Message', () => {

        render(
            <PageProvider>
                <TasksPage />
            </PageProvider>
        );

        expect( screen.getByText("You don't have any task in the list") ).toBeTruthy();
    });

    test('Input and Creation Test', () => {

        render(
            <PageProvider>
                <TasksPage />
            </PageProvider>
        );

        const input = screen.getByPlaceholderText('Type the task you need');

        fireEvent.input( input, { target: { value: 'Task 1 Test' } });

        expect( input.value ).toBe('Task 1 Test');

        const button = screen.getByRole('button');
        fireEvent.click( button );


        //Evaluate the success alert
        expect( screen.getByText("Success") ).toBeTruthy();
        expect( screen.getByText("To Do Task Saved.") ).toBeTruthy();

        //Evaluate the list item with active status
        expect( screen.getByText("Task 1 Test") ).toBeTruthy();
        expect( screen.getByText("Status: Active") ).toBeTruthy();
        expect( input.value ).toBe('');
    });

    test('Task Alert Deletion Test', () => {
        render(
            <PageProvider>
                <TasksPage />
            </PageProvider>
        );

        //Ensuring initial conditions
        const input = screen.getByPlaceholderText('Type the task you need');
        expect( screen.getByText("Task 1 Test") ).toBeTruthy();
        expect( screen.getByText("Status: Active") ).toBeTruthy();
        expect( input.value ).toBe('');

        //Delete action
        const button = screen.getByRole('button', { name: 'delete' });
        fireEvent.click( button );


        //Evaluate the warning Alert
        expect( screen.getByText("Are you Sure?") ).toBeTruthy();
        expect( screen.getByText("Do you want to delete this to do task ?") ).toBeTruthy();
        expect (screen.getByRole('button', {name:"Yes, I want to delete it."})).toBeTruthy();
    });
});