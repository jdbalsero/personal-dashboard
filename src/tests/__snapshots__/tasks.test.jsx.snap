// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TasksPage /> Tests Match Snapshot 1`] = `
<div>
  <div
    class="todo-general-container"
  >
    <h1
      class="ContentHeader"
      style="color: rgb(88, 115, 254); font-weight: normal;"
    >
      Tasks
    </h1>
    <div
      class="MuiBox-root css-8atqhb"
      style="display: flex; justify-content: center; margin: 17px;"
    >
      <span
        class="MuiTypography-root MuiTypography-body2 css-1phucj-MuiTypography-root"
      >
        You don't have any task in the list
      </span>
    </div>
    <div
      class="todo-add-button-container"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary inputTodo css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          placeholder="Type the task you need"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
      <button
        class="todo-add-button"
        type="button"
      >
        <div>
          Add Task
        </div>
      </button>
    </div>
  </div>
</div>
`;
